game-core/src/main/java/games/strategy/triplea/delegate/battle/end/NoMoreUnits.java

```diff
@@ -0,0 +1,77 @@
+package games.strategy.triplea.delegate.battle.end;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import lombok.Builder;
+import lombok.NonNull;
+import org.triplea.java.collections.CollectionUtils;
+
+@Builder
+public class NoMoreUnits {
+
+  public enum Winner {
+    ATTACKER,
+    DEFENDER,
+    DEFENDER_TURN_1,
+    UNDETERMINED
+  }
+
+  private @NonNull final GameData gameData;
+  private @NonNull final GamePlayer attacker;
+  private @NonNull final Collection<Unit> attackingUnits;
+  private @NonNull final Collection<Unit> defendingUnits;
+  private @NonNull final Territory battleSite;
+  private final int round;
```

Comment:
To force this value to be set with a builder, it would need to be an `Integer`, then you could use `@Nonnull` with it and enforce that it'll be specified when constructing otherwise this can silently default to zero.  Is it intentional to allow this to have a default?

Reply:
No, I wasn't aware that was what builder would do.  I'll fix that.